plenum/test/view_change/test_view_change_with_different_ic.py

```diff
@@ -0,0 +1,106 @@
+    3. nodes_to_restart send InstanceChanges for all nodes.
+    4. Ensure elections done.
+    """
+    nodes_to_restart = txnPoolNodeSet[1:3]
+    panic_node = txnPoolNodeSet[-1]
+    view_no = txnPoolNodeSet[0].viewNo
+
+    panic_node.view_changer.on_master_degradation()
+    for n in nodes_to_restart:
+        _restart_node(looper, txnPoolNodeSet, n, tconf, tdir, allPluginsPath)
+    nodes_to_restart = txnPoolNodeSet[1:3]
+
+    for n in nodes_to_restart:
+        n.view_changer.on_master_degradation()
+
+    def check():
+        assert panic_node.view_change_in_progress
+
+    looper.run(eventually(check))
+    ensureElectionsDone(looper=looper, nodes=txnPoolNodeSet)
+    ensure_all_nodes_have_same_data(looper, nodes=txnPoolNodeSet)
+
+    for node in txnPoolNodeSet:
+        assert node.viewNo == view_no
+
+
+def test_view_change_not_happen_if_ic_is_discarded(looper, txnPoolNodeSet,
+                                                   sdk_pool_handle,
+                                                   sdk_wallet_client,
+                                                   tconf, tdir, allPluginsPath):
+    """
+    1. panic_node (Delta) send InstanceChange for all nodes.
+    2. Restart nodes_to_restart (Beta, Gamma).
+    3. Wait OUTDATED_INSTANCE_CHANGES_CHECK_INTERVAL sec.
+    4. nodes_to_restart send InstanceChanges for all nodes.
+    5. Ensure elections done.
+    """
+    nodes_to_restart = txnPoolNodeSet[1:3]
+    panic_node = txnPoolNodeSet[-1]
+    view_no = txnPoolNodeSet[0].viewNo
+
+    panic_node.view_changer.on_master_degradation()
+    for n in nodes_to_restart:
+        _restart_node(looper, txnPoolNodeSet, n, tconf, tdir, allPluginsPath)
+    nodes_to_restart = txnPoolNodeSet[1:3]
+
+    # waiting to discard InstanceChange
+    looper.runFor(tconf.OUTDATED_INSTANCE_CHANGES_CHECK_INTERVAL)
+
+    for n in nodes_to_restart:
+        n.view_changer.on_master_degradation()
+
+    def check():
+        assert all(panic_node.view_changer.instanceChanges.has_inst_chng_from(view_no + 1, node.name)
+                   for node in nodes_to_restart)
+        assert not panic_node.view_changer.instanceChanges.has_inst_chng_from(view_no + 1, panic_node.name)
+
+    looper.run(eventually(check))
+    ensureElectionsDone(looper=looper, nodes=txnPoolNodeSet)
+    ensure_all_nodes_have_same_data(looper, nodes=txnPoolNodeSet)
+
+    for node in txnPoolNodeSet:
+        assert node.viewNo == view_no
+
+
+def _restart_node(looper, txnPoolNodeSet, node_to_disconnect, tconf, tdir,
+                  allPluginsPath):
+    idx = txnPoolNodeSet.index(node_to_disconnect)
+    disconnect_node_and_ensure_disconnected(looper,
+                                            txnPoolNodeSet,
+                                            node_to_disconnect)
+    looper.removeProdable(name=node_to_disconnect.name)
+
+    # add node_to_disconnect to pool
+    node_to_disconnect = start_stopped_node(node_to_disconnect, looper, tconf,
```

Comment:
Are we sure that this is enough to clear state of `node_to_disconnect`?

Reply:
Yes, beacause start_stopped_node create a new TestNode